cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/cruisecontrol/CruiseControlApiImpl.java

```diff
@@ -24,35 +34,98 @@
 
     private final Vertx vertx;
     private final long idleTimeout;
-
-    public CruiseControlApiImpl(Vertx vertx) {
-        this(vertx, HTTP_DEFAULT_IDLE_TIMEOUT_SECONDS);
+    private Secret ccSecret;
+    private Secret ccApiSecret;
+    private Secret coSecret;
+    private boolean apiAuthorizationEnabled;
+    private boolean apiAuthenticationEnabled;
+    private HTTPHeader authHttpHeader;
+
+    public CruiseControlApiImpl(Vertx vertx, Secret ccSecret, Secret ccApiSecret, Secret coSecret, Boolean apiAuthorizationEnabled, boolean apiAuthenticationEnabled) {
```

Comment:
Why not just call the constructor below with the `HTTP_DEFAULT_IDLE_TIMEOUT_SECONDS` as the `idleTimeout` argument, instead of rewriting all these assignments?

Reply:
You are right, done!